src/main/java/erb/RouteManager.java

```diff
@@ -142,11 +142,18 @@ private static void parseRouteObject(JSONObject route, ArrayList<Route> rList) {
 		//get route data
 		// erb: vk US-57-TASK-106 change node to routeName
 		String routeName = (String) routeObj.get("routeName");
-		String length = (String) routeObj.get("length");
+		// erb: vk US-57-TASK-117 added path
+		// erb: changed to correct types
+	    String path = (String) routeObj.get("path");
+	       Object tmpv = routeObj.get("length");
 
-		String duration = (String) routeObj.get("duration");
-		String routeId = (String) routeObj.get("routeId");
-		rList.add(new RouteImpl(routeName, length, duration, routeId));		
+	    double length = ((Double) routeObj.get("length")).intValue();
```

Comment:
this is effectively truncating the values and removing your default fractional parts of length/duration
```suggestion
	    double length = ((Double) routeObj.get("length"));
```

Reply:
Oh, whoops! Fixed.  :)